crates/interfaces/src/p2p/error.rs

```diff
@@ -1,14 +1,44 @@
 use crate::consensus;
-use reth_primitives::{rpc::BlockNumber, WithPeerId, H256};
+use reth_primitives::{rpc::BlockNumber, BlockHashOrNumber, Header, WithPeerId, H256};
 use thiserror::Error;
 use tokio::sync::{mpsc, oneshot};
 
+use super::headers::client::HeadersRequest;
+
 /// Result alias for result of a request.
 pub type RequestResult<T> = Result<T, RequestError>;
 
 /// Result with [PeerId]
 pub type PeerRequestResult<T> = RequestResult<WithPeerId<T>>;
 
+/// Trait used to validate requests
+pub trait RequestValidation {
+    /// Determine whether the response matches what we requested in request
+    fn is_likely_a_bad_message(&self, request: &HeadersRequest) -> bool;
+}
+
+impl RequestValidation for RequestResult<Vec<Header>> {
+    fn is_likely_a_bad_message(&self, request: &HeadersRequest) -> bool {
+        match self {
+            Ok(headers) => {
+                let request_length = headers.len() as u64;
+
+                if request_length < 1 && request.limit != request_length {
```

Comment:
```suggestion
                if request_length == 0 && request.limit > 0 {
```

Is this equivalent? What do we want to ensure here?

Reply:
Related to [this comment](https://github.com/paradigmxyz/reth/pull/946#discussion_r1083273973) above. Indeed I think your suggestion is equivalent and easier to read.

Reply:
ah my bad, meant to type `<=1` here

bad responses are usually empty or 1 but the requested blocks range was larger